assets/js/123validations.js

```diff
@@ -0,0 +1,62 @@
+function content_for_stepIsValid(clickedButton){
+  var stepIsValid = true;
+  var fieldsetId = clickedButton.data('fieldset');
+
+  var stepValues = $('#'+fieldsetId).serialize();
+  stepValues = serialize_unchecked(fieldsetId, stepValues);
+  stepValues = stepValues.split('&');
+
+  for( var i = 0; i < (stepValues.length); i++){
+    var keyValue = stepValues[i].split('=');
+    if(keyValue[0] != 'comments'){
+      var field = $('#'+keyValue[0]);
+      var value = keyValue[1].trim();
+      stepIsValid = (perform_dedicate_validation(field, value) && stepIsValid);
+    }
+  }
+  return stepIsValid;
+}
+
+function perform_dedicate_validation(field, value){
+  var fieldIsValid = true;
+  var validationType = field.data('validation-type');
+  var fieldId = field.attr('id');
+  switch(validationType) {
+    case 'drop_or_radio':
+      fieldId = field.data('parent-id');
+      fieldIsValid = ($('#'+fieldId).text() == $('#'+fieldId+'_id').val());
+      break;
+    case 'only_letters':
+      fieldIsValid = (value.length >= 10) && (/^[a-z\+]+$/i.test(value) );
+      break;
+    case 'phone':
+      fieldIsValid = (value.length >= 8) && !(value.match(/[^0-9]+/gi));
+      break;
+    case 'email':
+      fieldIsValid =  (value.length >= 6) && ((value.match(/[A-Z0-9._%+-]+%40[A-Z0-9.-]+\.[A-Z]{2,4}/gi)) != null);
+      break;
+    case 'mandatory_check':
+      fieldIsValid = (value == 'Si');
+      break;
+    default:
+      fieldIsValid = (value !== '');
+  }
+
+  fieldIsValid ? $('#'+fieldId+'_tooltip').hide() : $('#'+fieldId+'_tooltip').show();
+  return fieldIsValid;
+}
+
+function serialize_unchecked(step, stepValues){
+  var serialized = $('#'+step).serialize();
+    $('#'+step+' input:checkbox:not(:checked)').each(function(e){
+    stepValues += '&'+this.name+'=false';
+  })
+  return stepValues;
+}
+
+function hide_tooltip(){
```

Comment:
Identifier 'hide_tooltip' is not in camel case.<br>'hide_tooltip' is defined but never used.